### PR TITLE
docs: describe environment override feature

### LINKED ISSUE #21616: [Documentation Request] Environment config override
### Describe the feature

This was implemented here https://github.com/nuxt/nuxt/pull/20329 and added in the [blog](https://nuxt.com/blog/v3-5), but is currently not documented.

```ts
export default defineNuxtConfig({
  $production: {
    routeRules: {
      '/**': { isr: true }
    }
  },
  $development: {
    //
  }
})
```

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by adityarawat1223**
### 🔗 Linked issue

#21616 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I have Defined Newly added  Environment config override feature

Resolved #21616 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/1.getting-started/3.configuration.md [hunk 1]
```diff
@@ -29,6 +29,24 @@ Every configuration option is described in the [Configuration Reference](/docs/a
 You don't have to use TypeScript to build an application with Nuxt. However, it is strongly recommended to use the `.ts` extension for the `nuxt.config` file. This way you can benefit from hints in your IDE to avoid typos and mistakes while editing your configuration.
 ::
```
Comment (Hebilicious) [inline docs/1.getting-started/3.configuration.md:31]:
```suggestion

### Environment overrides

```

File: docs/1.getting-started/3.configuration.md [hunk 2]
```diff
@@ -29,6 +29,24 @@ Every configuration option is described in the [Configuration Reference](/docs/a
...
 
+You can configure fully type , per-environment overrides in your nuxt.config
+
```
Comment (Hebilicious) [inline docs/1.getting-started/3.configuration.md:33]: cc: @danielroe Could we suggest that these overrides are applied based on the key matching the `NODE_ENV` environment variable at dev time and build time (if that's how it actually works) ? 

Maybe explicitly show an example of how a config that has both $production and $development keys will be resolved ?

Comment (adityarawat1223) [reply docs/1.getting-started/3.configuration.md:35]: Thanks for suggestion i  accepted the changes now what to do ?

File: docs/1.getting-started/3.configuration.md [hunk 3]
```diff
@@ -29,6 +29,26 @@ Every configuration option is described in the [Configuration Reference](/docs/a
 You don't have to use TypeScript to build an application with Nuxt. However, it is strongly recommended to use the `.ts` extension for the `nuxt.config` file. This way you can benefit from hints in your IDE to avoid typos and mistakes while editing your configuration.
 ::
 
+### Environment overrides
+
+You can configure fully type , per-environment overrides in your nuxt.config
```
Comment (danielroe) [inline docs/1.getting-started/3.configuration.md:34]:
```suggestion
You can configure fully typed, per-environment overrides in your nuxt.config
```

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Thank you ❤️

**Comment by adityarawat1223** [comment]
Sorry if its a dumb question but is this issue done now ? can i move on to others ?

**Comment by danielroe** [comment]
Yes, this looks good to me. I'm going to go ahead and merge but it's possible @Hebilicious may have other ideas for improving the docs here?

**Comment by Hebilicious** [comment]
> Yes, this looks good to me. I'm going to go ahead and merge but it's possible @Hebilicious may have other ideas for improving the docs here?

Could we suggest that these overrides are applied based on the key matching the NODE_ENV environment variable at dev time and build time (if that's how it actually works) ?

Maybe explicitly show an example of how a config that has both $production and $development keys will be resolved ?

**Comment by danielroe** [comment]
I worry that mentioning NODE_ENV would be more confusing as it's not something that users should ever set themselves.

But perhaps saying that `development` is when running `nuxi dev`, while `nuxi build` and `nuxi generate` are production?

**Comment by danielroe** [comment]
Sounds good 👍

**Pull request merged**